pkg/autoops/api/api.go

```diff
@@ -412,15 +412,32 @@ func (s *AutoOpsService) validateDatetimeClauses(
 	clauses []*autoopsproto.DatetimeClause,
 	localizer locale.Localizer,
 ) error {
+	var checkClauses []*autoopsproto.DatetimeClause
```

Comment:
We could use map for checkClauses, which is faster and easier to understand code.